magicgui/core.py

```diff
@@ -326,14 +326,20 @@ class instance using setattr(self, self.WIDGET_ATTR.format(name), widget).
             try:
                 WidgetType = type2widget(arg_type)
             except TypeError:
+                funcname = getattr(self.func, "__name__", repr(self.func))
```

Comment:
not directly related to this PR ... but I found that `partial` functions weren't working since they don't have a `__name__` or `__annotation__` attribute.